### PR TITLE
hypershift: rosa hypershift mgmt

### PR DESCRIPTION
**Pull request by LiangquanLi930**
https://issues.redhat.com/browse/OCPQE-13447

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-rosa-sts-hypershift-mgmt-p2-f7

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-rosa-sts-hypershift-mgmt-p2-f7

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-rosa-sts-hypershift-mgmt-p2-f7

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-rosa-sts-hypershift-mgmt-p2-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-rosa-sts-hypershift-mgmt-p2-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-rosa-sts-hypershift-mgmt-p2-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-rosa-sts-hypershift-mgmt-p2-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-rosa-sts-hypershift-mgmt-p2-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-rosa-sts-hypershift-mgmt-p2-f7

**Comment by LiangquanLi930** [comment]
@kuiwang02  @liangxia Could you help to review this PR? thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable.yaml [hunk 1]
```diff
@@ -810,6 +810,21 @@ tests:
...
     workflow: cucushift-installer-rehearse-aws-ipi-workers-marketplace
+- as: aws-rosa-sts-hypershift-mgmt-p2-f7
+  cron: 14 3 6,13,22,30 * *
+  steps:
+    cluster_profile: aws-sd-qe
+    env:
+      CHANNEL_GROUP: candidate
+      OPENSHIFT_VERSION: "4.12"
+      TEST_ADDITIONAL: Hypershift
+      TEST_FILTERS: ~ChkUpgrade&;~DisconnectedOnly&;HyperShiftMGMT&;ROSA&
```
Comment (kuiwang02) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable.yaml:821]: @LiangquanLi930 better to `TEST_FILTERS: ~ChkUpgrade&;~DisconnectedOnly&;~MicroShiftOnly&;HyperShiftMGMT&;ROSA&`

File: ci-operator/step-registry/cucushift/hypershift-extended/enable-mgmt/cucushift-hypershift-extended-enable-mgmt-commands.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -euo pipefail
+
+if [ -f "${SHARED_DIR}/cluster-type" ] ; then
+    CLUSTER_TYPE=$(cat "${SHARED_DIR}/cluster-type")
+    if [[ "$CLUSTER_TYPE" == "osd" ]] || [[ "$CLUSTER_TYPE" == "rosa" ]]; then
+        cp "${SHARED_DIR}/kubeconfig" "${SHARED_DIR}/nested_kubeconfig"
+        cat "${SHARED_DIR}/hs-mc.kubeconfig" > "${SHARED_DIR}/kubeconfig"
```
[the remark this answers is not in the document]
Comment (LiangquanLi930) [reply ci-operator/step-registry/cucushift/hypershift-extended/enable-mgmt/cucushift-hypershift-extended-enable-mgmt-commands.sh:9]: yes, I debugged, no problem

[the remark this answers is not in the document]
Comment (LiangquanLi930) [reply ci-operator/step-registry/cucushift/hypershift-extended/enable-mgmt/cucushift-hypershift-extended-enable-mgmt-commands.sh:9]:
```
➜  export KUBECONFIG=kubeconfig
➜  oc get hostedcluster -A
NAMESPACE                                      NAME              VERSION                              KUBECONFIG                         PROGRESS    AVAILABLE   PROGRESSING   MESSAGE
ocm-staging-21oesg7g0p31rclerdf1skc4srv9o09t   xuelihp0208       4.12.1                               xuelihp0208-admin-kubeconfig       Completed   True        False         The hosted control plane is available
ocm-staging-21okte40chg3rm1qmk8j2nqj7jtqp6iv   sdq-ci-hovfg      4.12.1     
.......
```

[the remark this answers is not in the document]
Comment (kuiwang02) [reply ci-operator/step-registry/cucushift/hypershift-extended/enable-mgmt/cucushift-hypershift-extended-enable-mgmt-commands.sh:9]: ok, thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable.yaml (hunk 1 above)
Comment (LiangquanLi930) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-stable.yaml]: Thanks, updated

**Comment by LiangquanLi930** [comment]
@kuiwang02 Could you help to review again? Thanks!

**Comment by kuiwang02** [comment]
iamok from golang's view

**Comment by LiangquanLi930** [comment]
@liangxia Could you help to review this PR? thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml [hunk 3]
```diff
@@ -1152,6 +1152,21 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: osd-ccs-aws
+- as: aws-rosa-sts-hypershift-mgmt-p2-f2
+  cron: 28 17 2,5,7,9,11,13,15,17,19,21,23,25,27,29,31 * *
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:1156]: /cc @jianlinliu Is it fine to use `f2` ?

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:1156]: Ack

**Pull request merged**